src/System.Security.Cryptography.X509Certificates/src/Internal/Cryptography/Pal.Unix/StorePal.cs

```diff
@@ -150,7 +151,9 @@ public static IStorePal FromSystemStore(string storeName, StoreLocation storeLoc
             
             if ((openFlags & OpenFlags.ReadWrite) == OpenFlags.ReadWrite)
             {
-                throw new PlatformNotSupportedException(SR.Cryptography_Unix_X509_MachineStoresReadOnly);
+                throw new CryptographicException(
```

Comment:
Nah, in Windows and macOS we just pass through system codes, and on Linux it's entirely a .NET concept, so just these exceptions.